specs-go/state.go

```diff
@@ -33,3 +33,27 @@ type State struct {
 	// Annotations are key values associated with the container.
 	Annotations map[string]string `json:"annotations,omitempty"`
 }
+
+// FdIndexKey is the key used in the FdIndexes map of the ContainerProcessState struct.
+type FdIndexKey string
+
+const (
+	// SeccompFdIndexKey is the index of the seccomp notify file descriptor.
+	SeccompFdIndexKey FdIndexKey = "seccompFd"
+	// PidFdIndexKey is the index of the target process file descriptor.
+	PidFdIndexKey FdIndexKey = "pidFd"
+)
+
+// ContainerProcessState holds information about the state of a container process.
+type ContainerProcessState struct {
+	// Version is the version of the specification that is supported.
+	Version string `json:"ociVersion"`
+	// FdIndexes is a map containing the indexes of the file descriptors in the `SCM_RIGHTS` array.
+	FdIndexes map[FdIndexKey]int `json:"fdIndexes"`
```

Comment:
why isn't this just a normal array?

Reply:
Do you mean like this?

```suggestion
	Fds []FdKey `json:"fds"`
```

So it would look like this?
```json
{
    "ociVersion": "0.2.0",
    "fds": [
        "seccompFd",
        "pidFd"
    ],
    "pid": 4422,
    ...
}
```

I think that should work too.

The map structure comes from the previous implementation as an OCI hook. But I agree it's no longer necessary here.

Reply:
No, I don't think that will work, @alban . If tomorrow another optional fd is added (let's call it newFD), you don't have any way to know if the second element in the array is the pidFd or newFD.

@sargun for some context, in the previous iteration of this patch we had a top level field for each fd. Giuseppe asked to make this a map here (https://github.com/opencontainers/runtime-spec/pull/1074#discussion_r518651960).

Reply:
Ohh, sorry, I missed this was an array of strings. So, for the example I used it would be something like: ["seccompfd", "newFD"].

That can work too. Not sure about the type, though :)

Reply:
I updated the PR to use an array instead of a map.

Reply:
Thanks. Marking this as solved, then